pkg/gadgets/profile/tcprtt/tracer/tracer_test.go

```diff
@@ -0,0 +1,722 @@
+				err: true,
+			},
+		},
+		"filter_by_laddr_and_invalid_raddr": {
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterLocalAddress).Set(clientIP.String())
+				params.Get(ParamFilterRemoteAddress).Set("1.0.0.1")
+				return params
+			},
+			expectedResult: expectedResult{
+				err: true,
+			},
+		},
+		"filter_by_invalid_laddr_and_invalid_raddr": {
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterLocalAddress).Set("1.0.0.1")
+				params.Get(ParamFilterRemoteAddress).Set("1.0.0.1")
+				return params
+			},
+			expectedResult: expectedResult{
+				err: true,
+			},
+		},
+		"mix_by_local_and_filter_by_raddr": {
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamByLocalAddress).Set("true")
+				params.Get(ParamFilterRemoteAddress).Set(serverIP.String())
+				return params
+			},
+			expectedResult: expectedResult{
+				exactHistograms: 1,
+				byLocalAddr:     clientIP.String(),
+			},
+		},
+		"mix_by_local_and_filter_by_laddr": {
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamByLocalAddress).Set("true")
+				params.Get(ParamFilterLocalAddress).Set(clientIP.String())
+				return params
+			},
+			expectedResult: expectedResult{
+				exactHistograms: 1,
+				byLocalAddr:     clientIP.String(),
+			},
+		},
+		"mix_by_remote_and_filter_by_laddr": {
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamByRemoteAddress).Set("true")
+				params.Get(ParamFilterLocalAddress).Set(clientIP.String())
+				return params
+			},
+			expectedResult: expectedResult{
+				exactHistograms: 1,
+				byRemoteAddr:    serverIP.String(),
+			},
+		},
+		"mix_by_remote_and_filter_by_raddr": {
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamByRemoteAddress).Set("true")
+				params.Get(ParamFilterRemoteAddress).Set(serverIP.String())
+				return params
+			},
+			expectedResult: expectedResult{
+				exactHistograms: 1,
+				byRemoteAddr:    serverIP.String(),
+			},
+		},
+		// TODO: Test mix cases with clients and servers with different IPs
+	} {
+		test := test
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create context
+			gadgetCtx := newGadgetCtx(test.getGadgetParams(), test.timeout)
+			defer gadgetCtx.Cancel()
+
+			// Run the tracer in a goroutine
+			tracer := createTracer()
+			out := make(chan []byte)
+			go func() {
+				r, err := tracer.RunWithResult(gadgetCtx)
+
+				// Validate error. Use assert instead of require to avoid
+				// blocking the main goroutine.
+				if test.expectedResult.err {
+					assert.NotNil(t, err, "expected error running tracer")
+				} else {
+					assert.Nil(t, err, "not expected error running tracer")
+				}
```

Comment:
Another possibility is to pass a `struct {error, out}` to the channel and perform the whole validation on the "main" goroutine.